src/bindings/text/diff.rs

```diff
@@ -53,8 +58,23 @@ pub fn init(lua: &Lua) -> crate::Result<()> {
             format!("+++ {}\t{}", &right, time_format(&mtime(&right).map_err(LuaError::external)?))
         ];
 
-        let left = read_file(&left).map_err(LuaError::external)?;
-        let right = read_file(&right).map_err(LuaError::external)?;
+        if left == NULL_SOURCE && right == NULL_SOURCE {
+            return Err(rlua::Error::external(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "Both files cannot be null"
+            )));
+        }
+
+        let left = if left != NULL_SOURCE {
+            read_file(&left).map_err(LuaError::external)?
+        } else {
+            Vec::new()
+        };
+        let right = if right != NULL_SOURCE {
+            Vec::new()
```

Comment:
Should this be `read_file(&right).map_err(LuaError::external)?`?

Reply:
Fixed